### ISSUE TITLE
Invalid argument supplied for foreach thrown in EAV code 

### ISSUE THREAD

**Issue by wojtekn** [magento/magento2#21134]
### Preconditions

1. Latest Magento 2.3

I spot the problem in M2 Commerce in RMA module, but as problem comes from M2 Open Source core and is clearly visible by code review, I'm adding it here.

### Steps to reproduce

1. Enable RMA feature in backend
2. Place order in frontend
3. Invoice and ship order in backend
4. Navigate to order details in My Account pages in frontend
5. Open order details
6. Click "Return"

### Expected result

RMA form is displayed

### Actual result

Exception is thrown:

    1 exception(s):
    Exception #0 (Exception): Warning: Invalid argument supplied for foreach() in vendor/magento/module-rma/Block/Returns/Create.php on line 155

The problem looks as introduced in `\Magento\Eav\Model\Form::getAttributes` where not all class variables are initialized as array, but they are returned by method which suppose to return array:

    public function getSystemAttributes()
    {
        if ($this->_systemAttributes === null) {
            // load attributes
            $this->getAttributes();
        }
        return $this->_systemAttributes;
    }

`getAttributes` method initializes `_attributes` and `_userAttributes` variables as arrays but not `_systemAttributes`  and `_allowedAttributes`.

I will submit PR for this soon.

**Comment by magento-engcom-team** [comment]
Hi @wojtekn. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@wojtekn do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by riconeitzel** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @riconeitzel. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @riconeitzel, here is your [Magento instance](https://i-21134-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21134-2-3-develop.instances.magento-community.engineering/admin](https://i-21134-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98212`, `MAGETWO-98213` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by wojtekn** [comment]
@antbates1991 could you provide some insight on why it was closed?

**Comment by ghost** [comment]
Hi @antbates1991 please explain reason why you close this issue, we don't do like that just closing issue without reason.

**Comment by magento-engcom-team** [comment]
Hi @wojtekn. Thank you for your report. 
The issue has been fixed in magento/magento2#21135 by @wojtekn in 2.3-develop branch
Related commit(s):
 - [ac40d43e4a51ad235b4817d09f28cb50869c13a1](https://github.com/magento/magento2/commit/ac40d43e4a51ad235b4817d09f28cb50869c13a1)
 - [059f399232fe51f54e0e8c16986bf99c6fecbc4d](https://github.com/magento/magento2/commit/059f399232fe51f54e0e8c16986bf99c6fecbc4d)
 - [9f0dfcf648d8e657a5b8a1c4b231127a80a5a625](https://github.com/magento/magento2/commit/9f0dfcf648d8e657a5b8a1c4b231127a80a5a625)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @wojtekn. Thank you for your report. 
The issue has been fixed in magento/magento2#22086 by @wojtekn in 2.2-develop branch
Related commit(s):
 - [a1d6ed616b6d59c5238f102281fb36482a41acfd](https://github.com/magento/magento2/commit/a1d6ed616b6d59c5238f102281fb36482a41acfd)
 - [a7511e0bff4355b71bc21438ab333fcb04672e17](https://github.com/magento/magento2/commit/a7511e0bff4355b71bc21438ab333fcb04672e17)

The fix will be available with the upcoming 2.2.9 release.

**Comment by antbates91** [comment]
@wojtekn 
@engcom-backlog-nazar 

This issue was closed because I referenced this issue in a PR for private repository that has been merged into the private repos develop branch. This behavour was unexpected and in my opinion kind of wrong. If it can be configured it should probably be turned off, although I'm not sure if it is an option.

<img width="712" alt="Screenshot 2019-05-13 at 07 35 53" src="https://user-images.githubusercontent.com/8401438/57600698-79a2ef00-7552-11e9-8ae4-0f18d4def12a.png">
